src/Stage/InferTypes/GenerateEquations.elm

```diff
@@ -412,3 +456,165 @@ generateArgumentUsageEquations argumentId usages =
     List.map
         (Typed.getType >> equals argumentType)
         usages
+
+
+generatePatternEquations : Id -> Typed.LocatedPattern -> ( List TypeEquation, Id )
+generatePatternEquations currentId located =
+    let
+        ( pttrn, type_ ) =
+            Located.unwrap located
+    in
+    case pttrn of
+        Typed.PAnything ->
+            -- we can't make any assumptions here
+            ( [], currentId )
+
+        Typed.PVar _ ->
+            -- we can't make any assumptions here
+            ( [], currentId )
+
+        Typed.PRecord _ ->
+            -- we can't make any assumptions here
+            ( [], currentId )
+
+        Typed.PAlias pttrn_ _ ->
+            --TODO: not sure what to do here
```

Comment:
If this is `(<SOME OTHER PATTERN> as foo)`, perhaps we should delegate to `generatePatternEquations currentId <SOME OTHER PATTERN>`?

If there are issues with `generatePatternEquations` taking `LocatedPattern` instead of `Pattern` or `Pattern_` then perhaps we should split this whole function into two, and use the "unwrapped" one when delegating in this `PAlias` pattern.

Reply:
I'm also guessing the whole type will be the same as that delegated type (the var).
`equals type_ varType`